pkg/query-service/utils/format.go

```diff
@@ -3,11 +3,72 @@ package utils
 import (
 	"fmt"
 	"reflect"
+	"strconv"
 	"strings"
 
+	v3 "go.signoz.io/signoz/pkg/query-service/model/v3"
 	"go.uber.org/zap"
 )
 
+func ValidateAndCastValue(v interface{}, dataType v3.AttributeKeyDataType) (interface{}, error) {
+	switch dataType {
+	case v3.AttributeKeyDataTypeString:
+		switch x := v.(type) {
+		case string, int, float32, float64, bool:
+			return fmt.Sprintf("%v", x), nil
+		case []interface{}:
+			return v, nil
+		default:
+			return nil, fmt.Errorf("invalid data type, expected string, got %v", reflect.TypeOf(v))
+		}
+	case v3.AttributeKeyDataTypeBool:
+		switch x := v.(type) {
+		case bool:
+			return v, nil
+		case string:
+			boolean, err := strconv.ParseBool(x)
+			if err != nil {
+				return nil, fmt.Errorf("invalid data type, expected bool, got %v", reflect.TypeOf(v))
+			}
+			return boolean, nil
+		default:
+			return nil, fmt.Errorf("invalid data type, expected bool, got %v", reflect.TypeOf(v))
+		}
+	case v3.AttributeKeyDataTypeInt64:
+		switch x := v.(type) {
+		case int, int64:
+			return x, nil
+		case string:
+			int64val, err := strconv.ParseInt(x, 10, 64)
+			if err != nil {
+				return nil, fmt.Errorf("invalid data type, expected int, got %v", reflect.TypeOf(v))
+			}
+			return int64val, nil
+		default:
+			return nil, fmt.Errorf("invalid data type, expected int, got %v", reflect.TypeOf(v))
+		}
+	case v3.AttributeKeyDataTypeFloat64:
+		switch x := v.(type) {
+		case float32, float64:
+			return v, nil
+		case string:
+			float64val, err:= strconv.ParseFloat(x, 64)
+			if err != nil {
+				return nil, fmt.Errorf("invalid data type, expected float, got %v", reflect.TypeOf(v))
+			}
+			return float64val, nil
+		case int:
+			return float64(x), nil
+		case int64:
```

Comment:
can be a single line, int, int32, int64

Reply:
Then float64(x) is not working as x is treated a interface in that case